Doc/source/designspaceLib/xml.rst

```diff
@@ -53,6 +53,29 @@ Overview
             </dict>
         </lib>
     </designspace>
+    
+=====
+Types
+=====
+
+- **number**: A fixed point value `equivalent to Fixed in OpenType <https://docs.microsoft.com/en-us/typography/opentype/spec/otff#data-types>`_ (a 32-bit signed fixed-point number (16.16))
+- **font design (unsigned) float**: A floating point value `equivalent to (U)FWORD in OpenType <https://docs.microsoft.com/en-us/typography/opentype/spec/otff#data-types>`_
+- **float**: A floating point value `equivalent to F2DOT14 in OpenType <https://docs.microsoft.com/en-us/typography/opentype/spec/otff#data-types>`_
+
+     The F2DOT14 format consists of a signed, 2's complement integer and an unsigned fraction.
```

Comment:
This is an interesting precision but I wonder in practice how much success people have had with having their floating point values as written in e.g. designspace mappings be carried over to the font file with exactly the same value. Example practical use-case that comes up regularly: making a "staircase" mapping where a continuous change in user location maps to a brutal change in design location. How you would implement that is by making a mapping that has (for example):

| user | design |
| ---- | ---- |
| 100 | 1 |
| 199.9999 | 1 |
| 200 | 2 |
| 300 | 2 |

And ideally, the absolute perfect jump would be achieved by  having the "199.9999" value be exactly the F2DOT14 number that is before 200 as expressed in F2DOT14 (so that there's no opportunity for a ramp between the two numbers). Your addition of the precisions above to the spec makes me think that you're hinting at such a level of control; however I wonder how in practice that works out with compilers and such manipulating the floating point numbers, and whether the above is actually possible.

What use-case did you have in mind when mentioning such technical details as part of this format's documentation?

Reply:
Well, this PR stems from me trying to figure out minor issues in work that the talented @ewonrael did for the MFEK Foundation in [Grant №1 of 2022](https://mfek.org/foundation/). "MFEK Serif" is Besley.

He specified one of the axes as ending at 87.5, which I didn't know was even possible. I.e.:

https://github.com/indestructible-type/Besley/blob/c87ddc8b1d852033865cb52680b6bddb3da1b943/sources/designspace/Besley.designspace#L102
```xml
<dimension name="Width" xvalue="87.5"/>
```

That turned out not to be what's wrong, but I documented it for others.

Reply:
Ah OK, maybe is would be enough to say that floating point number are allowed, instead of referring to the particulars of the OT spec? I wonder how much one could argue that designspace is a "user facing format" that can/should be independent of the particular implementation of the OpenType format (which is also the only format really) but what if someone wants to make a designspace that is super-more-precise that F2DOT14 in its values, and uses a compiler or a font editor that doesn't care about OpenType? Maybe not a realistic scenario, but I'm not sure we should prevent it by tying the floating point values to the (limited) F2DOT14 format.

It's not even just about precision, but I realize now while writing that the F2DOT14 doesn't even go beyond the range of [-2,2], which is always exceeded by a Weight axis for example with values in the [0, 1000] range.